Show banner when sync fails due to quota error

AbeJellinek opened the pull request:
This handles (6) in #2376.

Also:
- Move strings to Fluent
- Fix overflow when window is narrow

Simulate a quota error locally for testing:
```diff
--- a/chrome/content/zotero/xpcom/storage/zfs.js
+++ b/chrome/content/zotero/xpcom/storage/zfs.js
@@ -449,6 +449,21 @@ Zotero.Sync.Storage.Mode.ZFS.prototype = {
 						debug: true
 					}
 				);
+				// Don't merge me! Simulate a quota error for testing:
+				Object.defineProperty(req, 'status', { value: 413 });
+				let getResponseHeader = ((original, obj) => ((header) => {
+					if (header === 'Zotero-Storage-Usage') {
+						return 999;
+					}
+					if (header === 'Zotero-Storage-Quota') {
+						return 1000;
+					}
+					if (header === 'Zotero-Storage-UserID') {
+						return Zotero.Users.getCurrentUserID();
+					}
+					return original.call(obj, header);
+				}))(req.getResponseHeader, req);
+				Object.defineProperty(req, 'getResponseHeader', { value: getResponseHeader });
 			}
 			catch (e) {
 				if (e instanceof Zotero.HTTP.UnexpectedStatusException) {
```

AbeJellinek committed:
Show banner when sync fails due to quota error

AbeJellinek committed:
Fix clipped banner in narrow window

dstillman commented on chrome/content/zotero/zoteroPane.js:
```diff
@@ -2839,16 +2830,17 @@ var ZoteroPane = new function()
 					setTimeout(() => {
 						this.showSetUpSyncReminder();
 						this.showAutoSyncReminder();
+						this.showQuotaErrorReminder();
```
I don't think this makes sense here. We show the `setUp` and `autoSync` reminders on item modification because it means you're making changes in your library, but that's not really an appropriate time to show the quota warning. I think we can just show it if 1) you hit a quota error and 2) you haven't said not to show the banner.

(Honestly, we should probably do the same for most unexpected, non-5xx sync errors — the idea being that a sync error should be dealt with and a banner is the way to communicate that. The icon then is just there so that there's still an indication that something is wrong if you dismiss the banner.)

So I think we could take out the `lastErrors` tracking and just show this immediately if not explicitly hidden.

dstillman commented on chrome/locale/en-US/zotero/zotero.ftl:
```diff
@@ -773,3 +774,16 @@ mac-word-plugin-install-remind-later-button =
+sync-reminder-autoSync-action = { general-enable }
+sync-reminder-quotaError-message = { $libraryType ->
+   *[user] You have reached your Zotero File Storage quota. Files you add may not sync.
+   [group] The owner of “{ $libraryName }” has reached their Zotero File Storage quota. Files in the group may not sync.
```
→ `Zotero Storage`

dstillman commented on chrome/content/zotero/zoteroPane.js:
```diff
@@ -2945,6 +2981,14 @@ var ZoteroPane = new function()
+						this.collectionsView.selectLibrary(options.library.libraryID);
+					}
+					else {
+						Zotero.launchURL('https://www.zotero.org/settings/storage');
```
We should put this in config.js, since we'll probably use it elsewhere.

dstillman commented on chrome/content/zotero/zoteroPane.js:
```diff
@@ -2945,6 +2981,14 @@ var ZoteroPane = new function()
 					break;
+				case 'quotaError':
+					if (options.library instanceof Zotero.Group) {
+						this.collectionsView.selectLibrary(options.library.libraryID);
```
I don't think there's much point to focusing the group. We should probably provide the owner `userID` in the server response so that the client could know if it's the current user. If it was, it could take you to your account settings, and otherwise it should probably just use the space to show a longer message that says to contact the group owner.

AbeJellinek committed:
Use Zotero-Storage-UserID

- Store remaining and userID on error
- We need to know the userID on later uploads before a new server-side
  quota error actually occurs, so replace storageRemainingForLibrary
  with lastQuotaErrorForLibrary and store error properties
- Split quota error reminder into two separate reminder types to reduce
  complexity in showSyncReminder()
